include/deal.II/base/mpi_compute_index_owner_internal.h

```diff
@@ -223,23 +260,31 @@ namespace Utilities
           ConsensusAlgorithmProcess(const IndexSet &owned_indices,
                                     const IndexSet &indices_to_look_up,
                                     const MPI_Comm &comm,
-                                    std::vector<unsigned int> &owning_ranks)
+                                    std::vector<unsigned int> &owning_ranks,
+                                    const bool track_index_requests = false)
             : owned_indices(owned_indices)
             , indices_to_look_up(indices_to_look_up)
             , comm(comm)
             , my_rank(this_mpi_process(comm))
             , n_procs(n_mpi_processes(comm))
+            , track_index_requests(track_index_requests)
             , owning_ranks(owning_ranks)
           {
-            this->dict.reinit(owned_indices, comm);
+            dict.reinit(owned_indices, comm);
+            requesters.resize(dict.actually_owning_rank_list.size());
           }
 
           const IndexSet &           owned_indices;
           const IndexSet &           indices_to_look_up;
           const MPI_Comm &           comm;
           const unsigned int         my_rank;
           const unsigned int         n_procs;
+          const bool                 track_index_requests;
           std::vector<unsigned int> &owning_ranks;
+          std::vector<std::vector<
+            std::pair<unsigned int,
+                      std::vector<std::pair<unsigned int, unsigned int>>>>>
+            requesters;
```

Comment:
Can you add a comment about what is the data stored here.

Reply:
Done!